### ISSUE TITLE
System.IO.DirectoryNotFoundException thrown when entering an incorrect template

### ISSUE THREAD

**Issue by joncfoo** [fsprojects/Fix#30]
```
> mono ~/src/local/Fix/temp/bin/Fix.exe new
Enter project name:
> Foobar
Enter project directory (relative to working directory):
> .
Choose a template:
 - aspwebapi2
 - classlib
 - console
 - fslabbasic
 - fslabjournal
 - pcl259
 - servicefabrichost
 - servicefabricsuavestateless
 - sln
 - suave
 - websharperserverclient
 - websharperspa
 - windows

> conole
Generating project...
Creating /home/jonathan/Projects/fsharp/foobar/Foobar

Unhandled Exception:
System.IO.DirectoryNotFoundException: Directory '/home/jonathan/src/local/Fix/temp/bin/templates/conole' not found.
  at System.IO.Directory.ValidateDirectoryListing (System.String path, System.String searchPattern, System.Boolean& stop) <0x7f938b14d260 + 0x0019c> in <filename unknown>:0 
  at System.IO.Directory.GetFileSystemEntries (System.String path, System.String searchPattern, FileAttributes mask, FileAttributes attrs) <0x7f938b14d4a0 + 0x00066> in <filename unknown>:0 
  at System.IO.Directory.GetFiles (System.String path, System.String searchPattern) <0x7f938b14caf0 + 0x00018> in <filename unknown>:0 
  at System.IO.Directory.GetFilesRecurse (System.String path, System.String searchPattern, System.Collections.Generic.List`1 all) <0x7f938b14cbb0 + 0x00024> in <filename unknown>:0 
  at System.IO.Directory.GetFiles (System.String path, System.String searchPattern, SearchOption searchOption) <0x7f938b14cb10 + 0x00080> in <filename unknown>:0 
  at Fake.FileHelper.CopyDir (System.String target, System.String source, Microsoft.FSharp.Core.FSharpFunc`2 filterFile) <0x40584700 + 0x0006a> in <filename unknown>:0 
  at Project.New (System.String projectName, System.String projectDir, System.String templateName, Boolean paket, Boolean fake) <0x40581800 + 0x003f7> in <filename unknown>:0 
  at Commands.project (Argu.ParseResults`1 results) <0x40576ea0 + 0x00a33> in <filename unknown>:0 
  at FixLib+handleInput@30-1.Invoke (Argu.ParseResults`1 results) <0x40576e70 + 0x00013> in <filename unknown>:0 
  at Commands.processCommand[T] (Microsoft.FSharp.Core.FSharpFunc`2 commandF, System.Object _arg1, System.String[] args) <0x40572c20 + 0x001e3> in <filename unknown>:0 
  at FixLib+handleInput@30.Invoke (Command arg10@, System.String[] args) <0x40572bd0 + 0x0003f> in <filename unknown>:0 
  at Microsoft.FSharp.Core.FSharpFunc`2[T,TResult].InvokeFast[V] (Microsoft.FSharp.Core.FSharpFunc`2 func, Microsoft.FSharp.Core.T arg1, Microsoft.FSharp.Core.TResult arg2) <0x40572b20 + 0x00074> in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.IO.DirectoryNotFoundException: Directory '/home/jonathan/src/local/Fix/temp/bin/templates/conole' not found.
  at System.IO.Directory.ValidateDirectoryListing (System.String path, System.String searchPattern, System.Boolean& stop) <0x7f938b14d260 + 0x0019c> in <filename unknown>:0 
  at System.IO.Directory.GetFileSystemEntries (System.String path, System.String searchPattern, FileAttributes mask, FileAttributes attrs) <0x7f938b14d4a0 + 0x00066> in <filename unknown>:0 
  at System.IO.Directory.GetFiles (System.String path, System.String searchPattern) <0x7f938b14caf0 + 0x00018> in <filename unknown>:0 
  at System.IO.Directory.GetFilesRecurse (System.String path, System.String searchPattern, System.Collections.Generic.List`1 all) <0x7f938b14cbb0 + 0x00024> in <filename unknown>:0 
  at System.IO.Directory.GetFiles (System.String path, System.String searchPattern, SearchOption searchOption) <0x7f938b14cb10 + 0x00080> in <filename unknown>:0 
  at Fake.FileHelper.CopyDir (System.String target, System.String source, Microsoft.FSharp.Core.FSharpFunc`2 filterFile) <0x40584700 + 0x0006a> in <filename unknown>:0 
  at Project.New (System.String projectName, System.String projectDir, System.String templateName, Boolean paket, Boolean fake) <0x40581800 + 0x003f7> in <filename unknown>:0 
  at Commands.project (Argu.ParseResults`1 results) <0x40576ea0 + 0x00a33> in <filename unknown>:0 
  at FixLib+handleInput@30-1.Invoke (Argu.ParseResults`1 results) <0x40576e70 + 0x00013> in <filename unknown>:0 
  at Commands.processCommand[T] (Microsoft.FSharp.Core.FSharpFunc`2 commandF, System.Object _arg1, System.String[] args) <0x40572c20 + 0x001e3> in <filename unknown>:0 
  at FixLib+handleInput@30.Invoke (Command arg10@, System.String[] args) <0x40572bd0 + 0x0003f> in <filename unknown>:0 
  at Microsoft.FSharp.Core.FSharpFunc`2[T,TResult].InvokeFast[V] (Microsoft.FSharp.Core.FSharpFunc`2 func, Microsoft.FSharp.Core.T arg1, Microsoft.FSharp.Core.TResult arg2) <0x40572b20 + 0x00074> in <filename unknown>:0
```

Perhaps instead of bombing out, it could a) loop back (retry) or b) use a string distance function like Levenshtein distance and just go ahead if it's accurate enough.

P.S. I've found other invalid/incorrect input-related issues in other states/commands (e.g. enter a blank line when at the template prompt).

Happy to work on it - lmk

**Comment by Krzysztof-Cieslak** [comment]
https://github.com/ionide/ionide-fsharp/blob/develop/src/Components/CompletionHelpers.fs#L72-L87 - nice implementation of Levenshtein which can be just copy pasted.

**Comment by reidev275** [comment]
Levenshtein would be cool but I'd be pretty happy to simply redisplay the list.  You guys are playing in advanced mode :+1:

**Comment by Krzysztof-Cieslak** [comment]
Fixed (a bit) in `0.3.4` - right now program does nothing (prints info) and ends if wrong template name is given.

We can come back to it later in the future, but closing for now.
